### DISCUSSION TITLE
Batch mode cliff processing error

### DISCUSSION THREAD

**Discussion by boojums** [karttapullautin/karttapullautin#124] [Q&A] [answered]
TLDR: I frequently get an indexing error during makecliffs in batch mode, but only with some adjacent files also present. 

Not 100% sure this is a bug and not some kind of user or data error, so I've put this here. Any insight is appreciated!

The error is:

thread 'main' panicked at src/cliffs.rs:259:43:
index out of bounds: the len is 335 but the index is 335
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic_bounds_check
   3: pullauta::cliffs::makecliffs
   4: pullauta::process::process_tile
   5: pullauta::process::batch_process
   6: pullauta::main

Here's an example file ([522519](https://rockyweb.usgs.gov/vdelivery/Datasets/Staged/Elevation/LPC/Projects/AZ_PimaCounty_2021_B21/AZ_PimaCo_2_2021/LAZ/USGS_LPC_AZ_PimaCounty_2021_B21_522519.laz)) that causes this error in batch mode, but only if one particular adjacent file ([522520](https://rockyweb.usgs.gov/vdelivery/Datasets/Staged/Elevation/LPC/Projects/AZ_PimaCounty_2021_B21/AZ_PimaCo_2_2021/LAZ/USGS_LPC_AZ_PimaCounty_2021_B21_522520.laz
)) is included in the batch input folder.

I do **not** get this error when I process the file:
- by itself as a command line option
- in batch mode as the only file in the batch input folder
- with other (adjacent or non-adjacent) files in the batch input folder, but NOT 522520. 

So, is it me, or is it a bug?

**Comment by rphlo** [comment, marked as answer]
There is an obvious error in the code where it crashes... I will fix it right away.

**Comment by rphlo** [comment]
Fixed in v2.0.5

**Reply by boojums** [reply]
Awesome, thanks! It's working so far!
